experimental/Schemes/SpaceGerms.jl

```diff
@@ -48,13 +58,141 @@ A space germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X``
   end
 end
 
+@doc raw"""
+    HypersurfaceGerm{BaseRingType, RingType, SpecType}
+A hypersurface germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X`` of type SpecType and local ring ``O_{(X,x)}`` of type `RingType` over some base ring ``k`` of type `BaseRingType`.
+"""
+@attributes mutable struct HypersurfaceGerm{
+                 BaseRingType<:Ring,
+                 RingType<:Ring,
+                 SpecType<:Spec} <: AbsSpaceGerm{BaseRingType, RingType}
+  X::SpecType
+  f::RingElem
+
+  function HypersurfaceGerm(X::GermAtClosedPoint,f::MPolyLocRingElem; check::Bool=true)
+    base_ring(modulus(OO(X))) == parent(f) || error("baserings do not match")
+    if check
+      (ideal(parent(f),[f]) == modulus(OO(X))) || error("given f does not define given X")
+    end
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X,f)
+  end
+
+## the following is currently unused....
+## as no backend for groebner computations is currently available in this case
```

Comment:
I think this is not true.

Reply:
ok. Tests added, as far as they make sense without Groebner bases. But the comment stays valid. We do not have anything building on top of this.